### DISCUSSION TITLE
Installing and running opensoundscape

### DISCUSSION THREAD

**Discussion by leahgray1** [kitzeslab/opensoundscape#652] [Q&A] [answered]
Hello,

Im very new to coding, so apologies in advance. I am trying to install and use opensoundscape, but I am running into some trouble. I have followed the instructions for downloading opensoundscape for microsoft (using ubuntu and anaconda) and it seems to install fine. However, Im not sure how you then get it to work in python. I have python 3.9, and when I try to import anything into python I get this error:

"ModuleNotFoundError: No module named 'opensoundscape'"

(Im just using lines from the tutorials:
from opensoundscape.audio import Audio
from opensoundscape.spectrogram import Spectrogram
)

Is there a step I am missing to import the module into python? or is it some sort of other issue e.g. path problems?

Thanks in advance for any help!

**Comment by sammlapp** [comment, marked as answer]
Hi @leahgray1 , I think the "missing step" here is selecting the Python environment you want to use. If you have followed the instructions [here](http://opensoundscape.org/en/latest/installation/windows.html), you created a "virtual environment" (perhaps called `opensoundscape` if you followed the tutorial exactly) and installed the opensoundscape package within that environment. To use the package, you'll need to use that environment in python. [This tutorial](https://towardsdatascience.com/getting-started-with-python-environments-using-conda-32e9f2779307) gives a decent beginner's intro to Python environments. 

Depending on how you are using python (command line or Jupyter Lab or something else) you will activate your Python environment in a slightly different way. To interact with Python in a notebook environment (similar to R notebooks, a combination of code, visuals, and text) take a look at [this part](http://opensoundscape.org/en/latest/installation/jupyter.html) of the OpenSoundscape documentation. In short, you need to let Jupyter know about the environment, for instance by running: 

`python -m ipykernel install --user --name opensoundscape ` 

then choose your environment from the drop-down menu or launch panel in JupyterLab. 

To simply run python in command line, just make sure to active your environment first (`conda activate opensoundscape`) before starting python by running `python` or `ipython`)

If you have further questions or think your issue is not related to choosing the correct environment, feel free to respond below!

**Reply by leahgray1** [reply]
Hello, 

Thanks alot for your help - Ive managed to get it up and running in Jupyter Lab.
